### ISSUE TITLE
Cannot get gemini model to return None using pydanticAI

### ISSUE THREAD

**Issue by DataMonsterBoy** [pydantic/pydantic-ai#533]
I am trying to create the return type of a pydantic model to be an integer or None from the prompt. However, setting the return as None causes an error with a gemini model

```py
from pydantic_ai import Agent
from pydantic import BaseModel
from pydantic_ai.models.vertexai import VertexAIModel

class AgeModel(BaseModel):
    age: int | None

gemini_model = VertexAIModel('gemini-1.5-flash')

prompt = "'The man was very old'. How old was the man? If you cannot find his age then don't return anything"
agent = Agent(gemini_model, result_type=AgeModel)
age = agent.run_sync(prompt)
```

```py
pydantic_ai.exceptions.UnexpectedModelBehavior: Unexpected response from gemini 400, body:
{
  "error": {
    "code": 400,
    "message": "Invalid value at 'tools.function_declarations[0].parameters.properties[0].value.any_of[1].type' (type.googleapis.com/google.cloud.aiplatform.v1.Type), \"null\"",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "tools.function_declarations[0].parameters.properties[0].value.any_of[1].type",
            "description": "Invalid value at 'tools.function_declarations[0].parameters.properties[0].value.any_of[1].type' (type.googleapis.com/google.cloud.aiplatform.v1.Type), \"null\""
          }
        ]
      }
    ]
  }
}
```

**Comment by SaidLopez** [comment]
I am getting the same using the gemini implementation option rather than Vertex

[/usr/local/lib/python3.10/dist-packages/pydantic_ai/models/gemini.py](https://localhost:8080/#) in _make_request(self, messages, streamed, model_settings)
    228             if r.status_code != 200:
    229                 await r.aread()
--> 230                 raise exceptions.UnexpectedModelBehavior(f'Unexpected response from gemini {r.status_code}', r.text)
    231             yield r
    232 

UnexpectedModelBehavior: Unexpected response from gemini 400, body:
{
  "error": {
    "code": 400,
    "message": "Invalid value at 'tools.function_declarations[0].parameters.properties[0].value.any_of[1].type' (type.googleapis.com/google.ai.generativelanguage.v1beta.Type), \"null\"",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "tools.function_declarations[0].parameters.properties[0].value.any_of[1].type",
            "description": "Invalid value at 'tools.function_declarations[0].parameters.properties[0].value.any_of[1].type' (type.googleapis.com/google.ai.generativelanguage.v1beta.Type), \"null\""
          }
        ]
      }
    ]
  }
}

**Comment by sydney-runkle** [comment]
Thanks for the report!

Huh, definitely seems gemini-specific. This works for anthropic:

```py
from pydantic_ai import Agent
from pydantic import BaseModel

class AgeModel(BaseModel):
    age: int | None


agent = Agent('claude-3-5-sonnet-latest', result_type=AgeModel)
age = agent.run_sync("'The man was very old'. How old was the man? If you cannot find his age then don't return anything")
print(age.data)
#> age=None
```

**Comment by samuelcolvin** [comment]
Thanks for reporting.

This should be mostly fixed by #540, although note that Gemini has pretty poor support for JSON schema, so you'll need to set a default of `None` on any optional field, e.g.

```py
class AgeModel(BaseModel):
    age: int | None = None  # <-- this will work

class AgeModel(BaseModel):
    age: int | None  # <-- this will not work
```

**Comment by DataMonsterBoy** [comment]
So the code now works although it never seems to return None. It seems to behave as though it is:
```
class AgeModel(BaseModel):
    age: int
```

e.g.
```
from pydantic_ai import Agent
from pydantic import BaseModel
from pydantic_ai.models.vertexai import VertexAIModel

class AgeModel(BaseModel):
    age: int | None = None

gemini_model = VertexAIModel('gemini-1.5-flash')

prompt = "'The man was very old'. How old was the man? If you cannot find his age then don't return anything"
agent = Agent(gemini_model, result_type=AgeModel)
age = agent.run_sync(prompt)
print(age.data)
#> age=100
```
